api/v1/storagecluster_types.go

```diff
@@ -105,6 +105,9 @@ type StorageClusterSpec struct {
 	DefaultStorageProfile string `json:"defaultStorageProfile,omitempty"`
 
 	StorageProfiles []StorageProfile `json:"storageProfiles,omitempty"`
+
+	// StorageClassNames contains the name of the storage classes created for the respective storage class types
+	StorageClassNames map[string]string `json:"storageClassNames,omitempty"`
```

Comment:
Cannot we add storageclass names to `ManagedResourcesSpec` Spec?

Reply:
Good point, probably we can do that and keep all such details regarding auxiliary resources together.
WDYT @iamniting @umangachapagain ?

Reply:
I gave another look at the API structure, and think this might not be the best option.
We have NFS SC's as well, that are not part of the `ManagedResourcesSpec` field, and the `ManagedResourcesSpec` also contains info about Ceph dashboard, toolbox and couple of other things. So, adding the `StorageClassNames` field to it might not sit well.

Reply:
Just a thought ManagedResouceSpec contains different fields for block, file,nfs, etc you can add one more field storageclassname to reach specs to ManageCephBlockPools, ManageCephFilesystems, etc? and for NFS am not sure why it was kept outside of ManagedResouceSpec but you can add storageclassname to the NFS spec as well? 

@iamniting @umangachapagain WDYT?

Reply:
I like the idea. Probably better in comparison to using map.

Reply:
Ok, let me update the code accordingly.